posts/2025-10-21-25.0.0.11-beta.adoc

```diff
@@ -0,0 +1,286 @@
+
+In this beta release, Open Liberty introduces the new **Model Context Protocol (MCP) Server 1.0**, which enables AI applications to seamlessly interact with business logic and access real-time data through discoverable tools. It also features a **Netty-based HTTP transport** as a beta preview, offering an internal performance upgrade for HTTP, WebSocket, JMS, and SIP communications with zero migration impact.
+
+// // // // // // // //
+// Change the RELEASE_SUMMARY to an introductory paragraph. This sentence is really
+// important because it is supposed to grab the readers attention.  Make sure to keep the blank lines
+//
+// Throughout the doc, replace 25.0.0.11-beta with the version number of Open Liberty, eg: 22.0.0.2-beta
+// // // // // // // //
+
+The link:{url-about}[Open Liberty] 25.0.0.11-beta includes the following beta features (along with link:{url-prefix}/docs/latest/reference/feature/feature-overview.html[all GA features]):
+
+* <<mcp, Model Context Protocol Server>>
+* <<netty, Netty‑based HTTP transport on Open Liberty>>
+
+See also link:{url-prefix}/blog/?search=beta&key=tag[previous Open Liberty beta blog posts].
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // // 
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/32991
+// Contact/Reviewer: habiblawal1,Azquelt
+// // // // // // // // 
+[#mcp]
+== Model Context Protocol Server
+
+link:https://modelcontextprotocol.io/docs/getting-started/intro[Model Context Protocol (MCP)] is an open standard to enable AI applications to interact with and utilize external systems.
+
+The new MCP Server 1.0 feature (`mcpServer-1.0`) allows developers of applications on Liberty to bridge the gap between LLMs and their business code by providing the capability to expose functionality for AI applications to discover, understand, and use.
+
+### Why use MCP Server
+
+Imagine your company provides weather forecasting services that require constant updates from multiple data sources. Since the weather forecast changes frequently, it's impossible for an LLM trained on data from last week to have much idea about the weather this week. A much better approach is to allow the LLM to access this weather data via a tool that your Liberty app exposes. The LLM can then call the tool whenever it needs forecasts and instantly have the up-to-date weather information.
+
+### How to use the MCP Server 1.0 feature
+To expose your business logic to authorized AI applications, you'll need to declare it as an MCP link:https://modelcontextprotocol.io/specification/2025-06-18/server/tools[tool]. In this context, a tool is simply a function or operation that the AI can invoke to perform a specific task.
+
+The Liberty `mcpServer-1.0` feature makes this simple by allowing you to declare a tool using the `@Tool` annotation on a method of a link:https://openliberty.io/guides/cdi-intro.html#managing-scopes-and-contexts[CDI managed bean] with a scope annotation.
```

Comment:
The Liberty mcpServer-1.0 feature makes this process simple by allowing you to declare a tool by using the @Tool annotation on a method of a CDI managed bean with a scope annotation.